automation/aws/aws_rbd_from_tag/aws_rbd_from_tag.pt

```diff
@@ -0,0 +1,265 @@
+  result "result"
+  code <<-EOS
+  result = []
+
+  _.each(ds_aws_accounts_with_tags, function(account) {
+    tags = {}
+
+    if (account['tags'] != null && account['tags'] != undefined) {
+      _.each(account['tags'], function(tag) {
+        key = tag['Key'].toLowerCase().trim()
+        value = tag['Value']
+        tags[key] = value
+      })
+    }
+
+    result.push({
+      arn: account['arn'],
+      id: account['id'],
+      name: account['name'],
+      status: account['status'],
+      tags: tags
+    })
+  })
+EOS
+end
+
+datasource "ds_rbds" do
+  run_script $js_rbds, $ds_aws_accounts, $ds_existing_rbds, $param_tag_list, $param_effective_date
+end
+
+script "js_rbds", type: "javascript" do
+  parameters "accounts", "existing_rbds", "param_tag_list", "param_effective_date"
+  result "result"
+  code <<-'EOS'
+  result = []
+
+  rbd_id_list = _.pluck(existing_rbds, 'id')
+
+  _.each(param_tag_list, function(tag) {
+    rbd_id = "rbd_" + tag.toLowerCase().replace(/\s+/g, '').replace(/\W/g, "").replace('.', '').replace('-', '').trim()
+    rbd_name = tag.replace('.', ' ').replace('-', ' ')
+    rbd_name = rbd_name.replace(/\W/g, " ").trim()
+    tag_key = tag.toLowerCase().trim()
+    verb = "POST"
+
+    if (_.contains(rbd_id_list, rbd_id)) {
+      verb = "PATCH"
+    }
+
+    rules = []
+
+    _.each(accounts, function(account) {
+      if (account['tags'][tag_key] != undefined && account['tags'][tag_key] != null) {
+        if (account['tags'][tag_key].trim() != '') {
+          rules.push({
+            "condition": {
+              "type": "dimension_equals",
+              "dimension": "vendor_account",
+              "value": account['id']
+            },
+            "value": {
+              "text": account['tags'][tag_key].toLowerCase().trim()
+            }
+          })
+        }
+      }
+    })
+
+    if (rules.length > 0) {
+      result.push({
+        id: rbd_id,
+        name: rbd_name,
+        verb: verb,
+        dated_rules: [
+          {
+            "effective_at": param_effective_date,
+            "rules": rules
+          }
+        ]
+      })
+    }
+  })
+EOS
+end
+
+datasource "ds_apply_rbds" do
+  iterate $ds_rbds
+  request do
+    run_script $js_apply_rbds, val(iter_item, "id"), val(iter_item, "verb"), val(iter_item, "name"), val(iter_item, "dated_rules"), rs_optima_host, rs_org_id
+  end
+  result do
+    encoding "text"
+  end
+end
+
+script "js_apply_rbds", type: "javascript" do
+  parameters "id", "verb", "name", "dated_rules", "rs_optima_host", "rs_org_id"
+  result "request"
+  code <<-EOS
+  request = {
+    auth: "auth_flexera",
+    host: rs_optima_host,
+    path: ["/bill-analysis/orgs/", rs_org_id, "/settings/rule_based_dimensions/", id].join(''),
+    verb: verb,
+    headers: {
+      "Api-Version": "1.0",
+      "content-type": "application/json"
+    },
+    body_fields: {
+      "name": name,
+      "dated_rules": dated_rules
+    }
+  }
+EOS
+end
```

Comment:
What was the reasoning for using DS to handle applying the resulting RBD rules ?

Reply:
bump @XOmniverse ^^  If these are being used we should consider merging them but wanted to sync again on this question firs